detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/out/HtmlOutputReport.kt

```diff
@@ -104,6 +115,16 @@ class HtmlOutputReport : OutputReport() {
             snippetCode(lineSequence, finding.startPosition, finding.charPosition.length())
         }
     }
+
+    private fun getComplexityMetrics(detektion: Detektion): List<String> {
+        var complexities = listOf<String>()
+        val complexityReportGenerator = ComplexityReportGenerator.create(detektion)
+        val complexityReport = complexityReportGenerator.generate()
+        return if (complexityReport.isNullOrBlank()) complexities else {
+            complexities = complexityReport.split("\n")
+            return complexities.subList(1, complexities.size - 1)
```

Comment:
```suggestion
            return complexityReport.split("\n").subList(1, complexities.size - 1)
```

The line above can be removed then. This way we spare the mutable variable.

Reply:
Not really. `complexities` is an empty list at this point and the above line populates it. Hence `complexities.size - 1` would be incorrect. 
We can either make `complexities`  as an immutable object or change it to following : 
`return complexityReport.split("\n").subList(1, complexityReport.split("\n").size - 1)`

Reply:
True. Then you could at least write something like this in order to move the mutable variable into the else clause.

```Kotlin
var complexities = complexityReport.split("\n")
return complexities.subList(1, complexities.size - 1)
```